src/pages/workspace/companyCards/WorkspaceCompanyCardsPage.tsx

```diff
@@ -35,10 +35,10 @@ function WorkspaceCompanyCardPage({route}: WorkspaceCompanyCardPageProps) {
     const selectedFeed = CardUtils.getSelectedFeed(lastSelectedFeed, cardFeeds);
     const [cardsList] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${workspaceAccountID}_${selectedFeed}`);
 
-    const companyCards = CardUtils.removeExpensifyCardFromCompanyCards(cardFeeds?.settings?.companyCards);
+    const companyCards = CardUtils.removeExpensifyCardFromCompanyCards(cardFeeds);
     const isLoading = !cardFeeds || !!(cardFeeds.isLoading && !companyCards);
-    const selectedCompanyCard = companyCards[selectedFeed ?? ''] ?? cardFeeds?.settings?.oAuthAccountDetails?.[selectedFeed ?? ''] ?? null;
-    const isNoFeed = isEmptyObject(companyCards) && isEmptyObject(cardFeeds?.settings?.oAuthAccountDetails) && !selectedCompanyCard;
+    const selectedCompanyCard = selectedFeed && companyCards[selectedFeed];
```

Comment:
I am confused by this property name, its the companyCards settings object, right? we should name it better
```suggestion
    const selectedCompanyCardSettings = selectedFeed && companyCards[selectedFeed];
```

Reply:
It's not really a setting, more like a data object. So I've renamed it to `selectedFeedData`, let me know if you feel differently.